common/arch/sdl/joy.cpp

```diff
@@ -466,5 +467,48 @@ int apply_deadzone(int value, int deadzone)
 		return 0;
 }
 
+bool joy_translate_menu_key(const d_event &event) {
+	if (event.type != EVENT_JOYSTICK_BUTTON_DOWN)
+		return false;
+#if DXX_MAX_BUTTONS_PER_JOYSTICK || DXX_MAX_HATS_PER_JOYSTICK || DXX_MAX_AXES_PER_JOYSTICK
+	// Find out which button has been pressed by decoding the button
```

Comment:
I dislike this approach.  If you need new data structures to track this properly, please add them.

Reply:
OK. Would a direct mapping from joystick buttons to key codes be acceptable? That would be the easiest way, I think.

Reply:
Yes.